zerver/views/messages.py

```diff
@@ -795,6 +796,42 @@ def json_update_message(request, user_profile):
     return update_message_backend(request, user_profile)
 
 @has_request_variables
+def highlight_message_differences(message_history, sender_email):
+    for edited_msg, prev_msg in zip(message_history, message_history[1:]):
+        # Iterate over message edit history
+        rendered_content = highlight_html_differences(prev_msg['prev_content'], \
+                                                      edited_msg['prev_content'])
+        edited_msg_index = message_history.index(edited_msg)
+        message_history[edited_msg_index]['prev_rendered_content'] = rendered_content
+        message_history[edited_msg_index]['edited_by'] = message_history[edited_msg_index+1]['edited_by']
+
+    # Add sender email as edited_by to original message
+    message_history[len(message_history)-1]['edited_by'] = sender_email
+    return message_history
+
+@has_request_variables
+def get_message_edit_history(request, user_profile,
+                              message_id=REQ(converter=to_non_negative_int)):
+    # type: (HttpRequest, UserProfile, int) -> HttpResponse
+    try:
+        message = Message.objects.get(id=message_id)
+    except Message.DoesNotExist:
+        return json_error(_("No such message"))
```

Comment:
We need to add a check here that the user in question has access to the message ID in question, or this will introduce a security problem where you can steal access to anyone's messages via this API endpoint!   

We should also add tests for this behavior.  I'm going to merge some relevant code for `fetch_raw_messages` that you can reuse for this purpose.